v2/clients/http/common_test.go

```diff
@@ -0,0 +1,81 @@
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package http
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/v2"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/clients/http/utils"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/dtos/common"
+)
+
+const (
+	TestUnexpectedMsgFormatStr   = "unexpected result, active: '%s' but expected: '%s'"
+	TestErrorMsgFromTargetClient = "received response: {%v}. error: %s"
+)
+
+func TestGetConfig(t *testing.T) {
+	ts := newTestServer(t, v2.ApiConfigRoute, common.ConfigResponse{})
+	defer ts.Close()
+
+	gc := NewCommonClient(utils.NewClient(ts.URL))
+	response, err := gc.Configuration(context.Background())
+	if err != nil {
+		t.Errorf(TestErrorMsgFromTargetClient, response, err)
+	}
+}
+
+func TestGetMetrics(t *testing.T) {
+	ts := newTestServer(t, v2.ApiMetricsRoute, common.MetricsResponse{})
+	defer ts.Close()
+
+	gc := NewCommonClient(utils.NewClient(ts.URL))
+	response, err := gc.Metrics(context.Background())
+	if err != nil {
+		t.Errorf(TestErrorMsgFromTargetClient, response, err)
+	}
+}
+
+func TestPing(t *testing.T) {
+	ts := newTestServer(t, v2.ApiPingRoute, common.PingResponse{})
+	defer ts.Close()
+
+	gc := NewCommonClient(utils.NewClient(ts.URL))
+	response, err := gc.Ping(context.Background())
+	if err != nil {
+		t.Errorf(TestErrorMsgFromTargetClient, response, err)
+	}
+}
+
+func TestVersion(t *testing.T) {
+	ts := newTestServer(t, v2.ApiVersionRoute, common.VersionResponse{})
+	defer ts.Close()
+
+	gc := NewCommonClient(utils.NewClient(ts.URL))
+	response, err := gc.Version(context.Background())
+	if err != nil {
+		t.Errorf(TestErrorMsgFromTargetClient, response, err)
+	}
+}
+
+func newTestServer(t *testing.T, apiRoute string, expectedResponse interface{}) *httptest.Server {
+	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		b, _ := json.Marshal(expectedResponse)
+		_, _ = w.Write(b)
+		if r.Method != http.MethodGet {
+			t.Errorf(TestUnexpectedMsgFormatStr, r.Method, http.MethodGet)
+		}
+		if r.URL.EscapedPath() != apiRoute {
+			t.Errorf(TestUnexpectedMsgFormatStr, r.URL.EscapedPath(), apiRoute)
+		}
```

Comment:
Put these first and return the proper HTTP error rather than falling the test. This will better simulate the HTTP request.

Reply:
done